python/ray/data/read_api.py

```diff
@@ -554,6 +555,70 @@ def read_mongo(
     )
 
 
+@PublicAPI(stability="alpha")
+def read_bigquery(
+    project_id: str,
```

Comment:
Is there a reason why `project_id` and `dataset` are flipped here vs. in the write_bigquery API?

Reply:
It seems from other conventions / hierarchy that `project_id` and then `dataset` should be the canonical way to do it :)

<img width="898" alt="Screen Shot 2023-09-28 at 3 53 46 PM" src="https://github.com/ray-project/ray/assets/113316/2ad36f95-08c4-4849-8e7d-1528198c2428">

Reply:
To clarify, when you say order, you mean the order of which the arguments are consumed?

The hierarchy is `project_id` is a higher level than `dataset`. I can make the arg order follow that if that's what you're requesting.

My intent was to list the args (required args first, then optional) in alphabetical order. For `read_bigquery()`, `dataset` is optional so `project_id` is first. For `write_bigquery()`, `dataset` and `project_id` are required so I listed dataset first.

Reply:
Given your information, the most natural order I believe would be
```
    def write_bigquery(
        self,
        project_id: str,
        dataset: str,
        ray_remote_args: Dict[str, Any] = None,
    )
```
and similar for the other write methods and keep read_bigquery as it is. Alphabetical ordering is not really a concern here, it is much more important that it is consistent between read and write, and with the hierarchy of concepts.

What do you think?

Reply:
Sounds good, I rearranged it to that order.